intake_esm/core.py

```diff
@@ -1,195 +1,151 @@
 
 from ._version import get_versions
-from .common import get_collection_def, get_subset, open_collection
-from .manage_collections import CESMCollections
+from .cesm import CESMCollection, CESMSource
+from .cmip import CMIPCollection, CMIPSource
+from .common import _get_built_collections, _open_collection
+from .config import INTAKE_ESM_CONFIG_FILE, SETTINGS
 
-__version__ = get_versions()["version"]
+__version__ = get_versions()['version']
 del get_versions
 
 
-class CesmMetadataStoreCatalog(Catalog):
-    """ CESM collection Metadata store """
+logger = logging.getLogger(__name__)
+logger.setLevel(level=logging.DEBUG)
 
-    name = "cesm_metadatastore"
+
+class ESMMetadataStoreCatalog(Catalog):
+    """ESM collection Metadata store. This class servers as an entry point for `intake_esm`. """
+
+    name = 'esm_metadatastore'
     version = __version__
+    collection_types = {'cesm': CESMCollection, 'cmip': CMIPCollection}
+    SOURCES = {'cesm': 'intake_esm.cesm.CESMSource', 'cmip': 'intake_esm.esm.CMIPSource'}
 
-    def __init__(self, collection, build_args={}, **kwargs):
+    def __init__(
+        self, collection_input_file=None, collection_name=None, collection_type=None, **kwargs
+    ):
         """
         Parameters
         ----------
-        collection : string
-                   CESM collection name
-        build_args : dict
-                    A dictionary containing arguments to enable and trigger a collection build.
-                    This dictionary should contain the following keys:
-
-                    - `collection_input_file`
-                    - `collection_type_def_file`
-                    - `overwriting_existing`
-                    - `include_cache_dir`
-                    For more info about these arguments, check :class:`~intake_cesm._manage_collections.CESMCollections`
 
-        """
-        self.collection = collection
-        self.build_args = build_args
-        if self.build_args:
-            self.df = self.build_collections()
-        else:
-            self.df = open_collection(collection)
+        collection_input_file : str,  Path, file
+                    Path to a YAML file containing collection metadata
+        collection_name : str
+                 Collection name
+
+        collection_type : str,
+                 Collection type. Accepted values include:
+
+                 - `cesm`
+                 - `cmip`
 
-        kwargs.setdefault("name", collection)
-        super(CesmMetadataStoreCatalog, self).__init__(**kwargs)
-        if self.metadata is None:
-            self.metadata = {}
-        self._entries = {}
 
-    def build_collections(self):
-        """ Build CESM collection
         """
+        self.collections = {}
+        self.get_built_collections()
 
-        if "collection_input_file" in self.build_args:
-            collection_input_file = self.build_args.get("collection_input_file")
+        if collection_name and collection_type:
```

Comment:
What if `collection_name and collection_type` *and* `collection_input_file`?

Reply:
Good point. How about raising an exception?

Reply:
I addressed this (probably not in the most elegant way :)). If you have suggestions for better alternative, let me know

Reply:
I thought about this and haven't come up with a clear alternative. I could adding a class method for building the collection from a file, and pass thru to this method when we detect that a file name is provided...but it's hard to know that a string is actually a file. I don't want to require the user to pass in a file object.